src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,31 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+  health = 100;
+  constructor(name) {
+    this.name = name;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (target instanceof Herbivore && target.hidden !== true) {
+      target.health -= 50;
+
+      if (target.health <= 0) {
+        Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
[CODE PERFORMANCE] - The current implementation of filtering `Animal.alive` to remove dead animals is done inside a conditional check which is nested inside the `bite` method. To enhance code performance and avoid nesting, consider filtering `Animal.alive` directly without the conditional check for each bite. This will avoid unnecessary checks and potential performance issues when dealing with a large number of animals.